### PR TITLE
feat(upload): Added post to download function

### PR DESCRIPTION
**Pull request by enri90**
Added post functionality to the download function

Example
```typescript
const body = JSON.stringify({ ids: ids });

await download(
  url,
  path,
  ({ progress, total }) => console.log(`Downloaded ${progress} of ${total} bytes`),
  new Map<string, string>([
    ['Content-Type', 'application/json'],
  ]),
  body
);
```

### CODE REVIEW

File: plugins/upload/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "tauri-plugin-upload"
-version = "2.0.0-beta.5"
+version = "2.0.0-beta.6"
```
Comment (FabianLars) [inline plugins/upload/Cargo.toml]: revert this please. Our publishing pipeline will do this.

File: plugins/upload/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@tauri-apps/plugin-upload",
-  "version": "2.0.0-beta.4",
+  "version": "2.0.0-beta.5",
```
Comment (FabianLars) [inline plugins/upload/package.json]: same

File: plugins/upload/guest-js/index.ts [hunk 3]
```diff
@@ -60,6 +61,7 @@ async function download(
     filePath,
     headers: headers ?? {},
     onProgress,
+    body: body ?? "",
```
Comment (FabianLars) [inline plugins/upload/guest-js/index.ts]: I think it's fine to just send null and use `Option<&str>` in Rust. Feels a bit cleaner but i don't really care that much.

File: plugins/upload/src/lib.rs [hunk 4]
```diff
@@ -66,10 +66,14 @@ async fn download(
     file_path: &str,
     headers: HashMap<String, String>,
     on_progress: Channel,
+    body: &str,
 ) -> Result<()> {
     let client = reqwest::Client::new();
-
-    let mut request = client.get(url);
+    let mut request = if !body.is_empty() {
+        client.post(url).body(body.to_owned())
```
Comment (FabianLars) [inline plugins/upload/src/lib.rs]: If we need an owned body var then you can just use `body: String` as the function argument.

**Review by FabianLars — COMMENTED** [review]
Thank you for contributing :) I added a few minor comments.

We'll also need a changefile (see the .changes folder for examples).

Do you think it makes sense to have the http request type configurable? Or should get and post cover it good enough for now

File: plugins/upload/guest-js/index.ts (hunk 3 above)
Comment (enri90) [reply plugins/upload/guest-js/index.ts]: I tried with Option< String > in Rust but it gives me problems on condition

**Comment by enri90** [comment]
> Thank you for contributing :) I added a few minor comments.
> 
> We'll also need a changefile (see the .changes folder for examples).
> 
> Do you think it makes sense to have the http request type configurable? Or should get and post cover it good enough for now

I need the post to send data when generating a PDF

**Pull request closed without merging**